### ISSUE TITLE
Cannot access TLS ConnectionState from a http.Request

### ISSUE THREAD

**Issue by MicOestergaard** [lucas-clemente/quic-go#2879]
In http3/server.go some information is copied from the quic.Session to the request in the method handleRequest. This allows one to access the data, e.g. RemoteAddr, in the ServeHTTP handler.

Unfortunately the request.TLS is not populated this way, so there is no way to access the quic.Session.ConnectionState() from a request. This means that there is no way to access a client certificate from a request handler.

A simple fix is to populate the request.TLS with data from the quic session in handleRequest.

**Comment by marten-seemann** [comment]
I thought we had fixed that a while ago, but I might be wrong. If so, a PR would be appreciated :)

**Comment by marten-seemann** [comment]
The problem here is that we can't fill out this field for 0-RTT requests, because `ConnectionState()` blocks until the handshake is completed.

**Comment by superfashi** [comment]
maybe we can expose a `HandshakeComplete()` to `quic.Session` interface so we can case this?

**Comment by superfashi** [comment]
In fact I can't seem to reproduce 0-RTT requests even with the current version, here's my code:

```go
go func() {
    log.Fatal(http3.ListenAndServeQUIC("localhost:443", "some.crt", "some.key", nil))
}()

c := http.Client{Transport: &http3.RoundTripper{
    TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
}}
request, _ := http.NewRequest(http3.MethodGet0RTT, "https://localhost:443", nil)
_, err := c.Do(request)
if err != nil {
    panic(err)
}
```

and it doesn't look like 0-RTT from wireshark, also the internal stream says the handshake is completed.

**Comment by ShivanshVij** [comment]
Is there an update for this? I've got a potential fix here: https://github.com/loopholelabs/quic-go/commit/ac0f32cf950c585695fc03693d2725d55bac658c but it seems too simple...

**Comment by marten-seemann** [comment]
No updates unfortunately.

> I've got a potential fix here: https://github.com/loopholelabs/quic-go/commit/ac0f32cf950c585695fc03693d2725d55bac658c but it seems too simple...

Unfortunately, this won't work with 0-RTT. `ConnectionState()` blocks until the handshake is complete.

**Comment by superfashi** [comment]
> No updates unfortunately.
> 
> > I've got a potential fix here: [loopholelabs@ac0f32c](https://github.com/loopholelabs/quic-go/commit/ac0f32cf950c585695fc03693d2725d55bac658c) but it seems too simple...
> 
> Unfortunately, this won't work with 0-RTT. `ConnectionState()` blocks until the handshake is complete.

I have a fix here which should work with 0-RTT https://github.com/superfashi/quic-go/commit/ab6736a34735da5bdce6180310335df0747b4f71

**Comment by marten-seemann** [comment]
@superfashi This means that `ConnectionState` won't be set when called before the handshake completes, right?
That's definitely not ideal, but maybe it's better than not setting it at all. We definitely need to carefully document this behavior.

Do you want to create a PR with your fix (I think we can just check for `Err() == nil`)?
